src/Microsoft.FeatureManagement/DynamicFeatureManager.cs

```diff
@@ -0,0 +1,253 @@
+            }
+            //
+            // IContextualFeatureVariantAssigner
+            else if (useAppContext &&
+                     TryGetContextualFeatureVariantAssigner(featureDefinition.Assigner, typeof(TContext), out ContextualFeatureVariantAssignerEvaluator contextualAssigner))
+            {
+                variant = await contextualAssigner.AssignVariantAsync(context, appContext, cancellationToken).ConfigureAwait(false);
+            }
+            //
+            // The assigner doesn't implement a feature variant assigner interface capable of performing the evaluation
+            else
+            {
+                throw new FeatureManagementException(
+                    FeatureManagementError.InvalidFeatureVariantAssigner,
+                    useAppContext ?
+                        $"The feature variant assigner '{featureDefinition.Assigner}' specified for the feature '{feature}' is not capable of evaluating the requested feature with the provided context." :
+                        $"The feature variant assigner '{featureDefinition.Assigner}' specified for the feature '{feature}' is not capable of evaluating the requested feature.");
+            }
+
+            if (variant == null)
+            {
+                variant = defaultVariant;
+            }
+
+            return await _variantOptionsResolver.GetOptionsAsync<T>(featureDefinition, variant, cancellationToken).ConfigureAwait(false);
+        }
+
+        private IFeatureVariantAssignerMetadata GetFeatureVariantAssignerMetadata(string assignerName)
+        {
+            const string assignerSuffix = "assigner";
+
+            IFeatureVariantAssignerMetadata assigner = _assignerMetadataCache.GetOrAdd(
+                assignerName,
+                (_) => {
+
+                    IEnumerable<IFeatureVariantAssignerMetadata> matchingAssigners = _variantAssigners.Where(a =>
+                    {
+                        Type assignerType = a.GetType();
+
+                        string name = ((AssignerAliasAttribute)Attribute.GetCustomAttribute(assignerType, typeof(AssignerAliasAttribute)))?.Alias;
+
+                        if (name == null)
+                        {
+                            name = assignerType.Name;
+                        }
+
+                        return IsMatchingMetadataName(name, assignerName, assignerSuffix);
+                    });
+
+                    if (matchingAssigners.Count() > 1)
+                    {
+                        throw new FeatureManagementException(FeatureManagementError.AmbiguousFeatureVariantAssigner, $"Multiple feature variant assigners match the configured assigner named '{assignerName}'.");
+                    }
+
+                    return matchingAssigners.FirstOrDefault();
+                }
+            );
+
+            return assigner;
+        }
+
+        private static bool IsMatchingMetadataName(string metadataName, string desiredName, string suffix)
```

Comment:
The comments in this function are still talking about filters. Is there a place we can put this helper function to so it can be shared?

Reply:
Sure. I put it in a new helper, `NameHelper`.